packages/pocket-ic/tests/test_canister.did

```diff
@@ -1,3 +1,32 @@
+type HeaderField = record {
```

Comment:
What do these types/endpoint enable?

Reply:
They enable using the [HTTP gateway protocol](https://internetcomputer.org/docs/current/references/http-gateway-protocol-spec) which is used when a browser (or `reqwest::Client`) connects to the HTTP gateway (e.g., `https://ic0.app` served by a BN) and the HTTP gateway translates the browset request into query/update calls uses those types/endpoints.

Reply:
I see, I guess I was just confused by the `Gateway` in the name `HttpGatewayRequest`, in other canisters i always see it as just `HttpRequest`. Thanks for clarifying!

Reply:
I used `Gateway` to disambiguate since I also import `HttpRequest` from the Rust CDK for canister http outcalls.